pytential/symbolic/primitives.py

```diff
@@ -858,7 +858,7 @@ def _small_mat_eigenvalues(mat):
         det_mat = a * d - b * c
 
         # solutions to lambda**2 - tr(A) * lambda + det(A)
-        sqrt_discriminant = cse(sqrt(tr_mat**2 - 4*det_mat))
+        sqrt_discriminant = cse(sqrt(abs(tr_mat**2 - 4*det_mat)))
```

Comment:
- Is it better to `abs` or to saturate at 0? (My gut feeling is saturate.)

Reply:
Or maybe specialize to a symmetric eigensolve?

Reply:
Went with specializing for a symmetric matrix, in which case that whole thing can be written as a sum of squares and is always nicely positive.

I also checked where it was failing, and there were just 3 nodes that had a value of `-4.4e-16`, so definitely some floating point issues.

Reply:
Thanks for checking! That makes me less worried about this whole affair. And the symmetric eigenvalue thing seems like a reasonably clean and reliable way out of this.